cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -81,8 +82,8 @@
   private final Map<String, LocalizeResource> resourcesToLocalize;
   private final Supplier<MapReduceTaskContextProvider> taskContextProviderSupplier;
 
-  private InputFormatProvider inputFormatProvider;
-
+  private Map<String, InputFormatProvider> inputFormatProviders;
+  private HashMap<String, Class<? extends Mapper>> mapperClasses;
```

Comment:
We can keep track of the InputFormatProvider and the Mapper Class in separate columns, where the row is the input name.